client/app/(auth)/createProfile.tsx

```diff
@@ -1,19 +1,430 @@
+  const currentUser = auth.currentUser;
+
+  const dispatch = useAppDispatch();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedAvatarOption, setSelectedAvatarOption] =
+    useState<AvatarOption>(avatarOptions[0]);
+  const [showBottomDrawer, setShowBottomDrawer] = useState(false);
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [selectedDate, setSelectedDate] = useState<Date>();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<UserEditModel>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      avatarUrl: "",
+      dateOfBirth: undefined,
+      username: "",
+      name: "",
+    },
+    mode: "onSubmit",
+  });
+
+  useEffect(() => {
+    const setDefaultUserData = async () => {
+      if (currentUser) {
+        const userData = await UserController.getUser(currentUser.uid);
+        if (userData) {
+          const matchedAvatar = avatarOptions.find(
+            (option) => option.url === userData!.avatarUrl,
+          );
+          if (matchedAvatar) {
+            setSelectedAvatarOption(matchedAvatar);
+          }
+
+          reset({
+            avatarUrl: userData.avatarUrl,
+            dateOfBirth: userData.dateOfBirth
+              ? new Date(userData.dateOfBirth)
+              : undefined,
+            name: userData.name,
+            username: userData.username,
+          });
+        }
+      }
+    };
+
+    setDefaultUserData();
+  }, [currentUser, reset]);
+
+  const handleCreateProfile = async (model: UserEditModel) => {
+    setIsLoading(true);
+    try {
+      if (currentUser) {
+        await UserController.editUser(currentUser.uid, model);
+        const userData = await UserController.getUser(currentUser.uid);
+        dispatch(setUser(userData));
+        router.push("/(auth)/createTeam");
+      } else {
+        router.push("/(auth)");
+      }
+    } catch (error) {
+      console.log(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
   return (
-    <View className="flex-1 items-center justify-center p-5">
-      <Text className="pbk-h6 text-gray-950">This is temporary</Text>
-      <Text className="pbk-b1 text-gray-950">{user.userId}</Text>
-      <Text className="pbk-b1 text-gray-950">{user.email}</Text>
-      <Text className="pbk-b1 text-gray-950">{user.username}</Text>
-
-      <Link className="mt-4 py-4" href="/">
-        <Text className="pbk-b2 text-purple-500">Go to home screen!</Text>
-      </Link>
+    <View className="flex-1 bg-gray-950">
+      <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+        <View className="flex-1">
+          <KeyboardAvoidingView behavior="padding" className="flex-1">
+            <View className="flex-1 flex-col px-6 py-4">
+              <View className="my-8 size-8 self-end" />
+
+              <Text className="pbk-h5 mb-8 text-base-white">
+                Tell us who you are
+              </Text>
```

Comment:
Instead of having that spacer View component I think we should just add a bit of marginTop to the Text. `mt-24` replicates the spacing you have set up on my screen. Let me know if this works, or if there is another use I am missing for this View component.

```suggestion
              <Text className="pbk-h5 mb-8 mt-24 text-base-white">
                Tell us who you are
              </Text>
```

Reply:
I agree, I don't like the extra space on the top. I would rather the content of the form start at the top with a bit of padding between the top of the screen, and then let it be as tall as it needs. Any extra white space should be pushed to in between the form and the continue button, similar to how it is in Nasrif's design

Reply:
Ok, I changed it now so the `View` spacer is removed and the margin is moved to the `Text` component. I reduced it a bit to `16px` so move it closer to the top of the screen.